### PR TITLE
move ownedbytes to own crate

### LINKED ISSUE #1106: Moved OwnedBytes to an external crate
(OwnedBytes would be useful in quickwit as well.)

### PR DESCRIPTION
**Pull request by PSeitz**
fixes #1106

### CODE REVIEW

File: ownedbytes/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            self.as_slice()
+        };
+        write!(f, "OwnedBytes({:?}, len={})", bytes_truncated, self.len())
+    }
+}
+
+impl Deref for OwnedBytes {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
```
Comment (fulmicoton) [inline ownedbytes/src/lib.rs:131]: should we inline this guy too?

File: ownedbytes/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                io::ErrorKind::UnexpectedEof,
+                "failed to fill whole buffer",
+            ));
+        }
+        Ok(())
+    }
+}
+
+impl AsRef<[u8]> for OwnedBytes {
+    fn as_ref(&self) -> &[u8] {
```
Comment (fulmicoton) [inline ownedbytes/src/lib.rs:179]: inline?

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Do we need to add the inline I suggested? Youare the expert

**Review by fulmicoton — APPROVED** [review]
Do we need to add the inline I suggested? Youare the expert

File: ownedbytes/src/lib.rs (hunk 1 above)
Comment (PSeitz) [reply ownedbytes/src/lib.rs:131]: yes, good catch

File: ownedbytes/src/lib.rs (hunk 2 above)
Comment (PSeitz) [reply ownedbytes/src/lib.rs:179]: added

**Pull request merged**